docs2/site/docs/guides/known-issues.md

```diff
@@ -346,3 +293,112 @@ Also see discussion in [#1310](https://github.com/graphql-dotnet/graphql-dotnet/
+> schema. Please use the same instance for all references within the schema, or use GraphQLTypeReference to reference a
+> type instantiated elsewhere.
+
+This is caused when the same graph type class has multiple distinct instances used within the schema. This is a change
+in GraphQL.NET v7, and exists as protection to ensure that the schema is initialized properly. Note that this restriction
+does not apply to scalars.
+
+Below are some samples of what does not work in GraphQL.NET v7, along with solutions:
+
+```csharp
+// sample 1: manually creating instances without providing the same instance for each use
+public class MyGraphType : ObjectGraphType
+{
+    public MyGraphType()
+    {
+        // creates an instance of WidgetGraphType
+        Field("field1", new WidgetGraphType());
+        // creates another instance of the same class
+        Field("field2", new WidgetGraphType());
+
+        // solution:
+        Field<WidgetGraphType>("field1");
+        Field<WidgetGraphType>("field2");
+    }
+}
+
+// sample 2: adding an instance to a union graph type
+public class MyUnionGraphType : UnionGraphType
+{
+    public MyUnionGraphType()
+    {
+        // creates an instance, which will be different than the one use elsewhere
+        AddPossibleType(new WidgetGraphType());
+
+        // solution:
+        Type<WidgetGraphType>();
+    }
+}
+
+// sample 3: pulling an instance from DI
+public class MyUnionGraphType : UnionGraphType
+{
+    public MyUnionGraphType(WidgetGraphType widgetType)
+    {
+        // since graph types are typically registered as transients, this reference to
+        // WidgetGraphType will be different than other instances throughout the schema
+        // and the following code does not work
+        AddPossibleType(widgetType);
+        ResolveType = obj => obj switch
+        {
+            string => widgetType,
+            _ => null,
+        };
+
+        // solution 1: register WidgetGraphType as a singleton manually with the DI provider
+
+        // solution 2: remove WidgetGraphType from the constructor and use the following code:
+        Type<WidgetGraphType>();
+        ResolveType = obj => obj switch
+        {
+            string => new GraphQLTypeReference("Widget"), // reference by name
+            _ => null,
+        };
+    }
+}
+```
+
+However, within your schema, you may pull the query, mutation and/or subscription types from DI.
+This is normal as those types are not typically referenced anywhere else in the schema.
+
+```csharp
+public class MySchema : Schema
+{
+    // correct implementation for the schema class
+    public MySchema(IServiceProvider serviceProvider, MyQueryGraphType queryGraphType)
+        : base(serviceProvider)
+    {
+        Query = queryGraphType;
+    }
+}
+```
+
+Similar restrictions apply if creating a dynamic schema. You will need to maintain a list of
```

Comment:
I think that it was inspired by issue from @OpenSpacesAndPlaces

Reply:
Yes, and if we encounter further samples from users, we should add them here.  We will not be able to anticipate all coding patterns in users' codebases.

Reply:
I have dynamic schema in my own codebases.  Usually I use generics, however.

Reply:
I have dynamic schema with dynamically generated (C# emit) "stub" scalar types. A niche case.